bench_runner/plot.py

```diff
@@ -75,21 +76,46 @@ class Options:
 
 
 @functools.cache
-def get_plot_config():
-    content = mconfig.get_bench_runner_config().get("plot", {})
-
-    for key in ["bases", "runners", "names", "colors", "styles", "versions", "markers"]:
-        assert key in content
-    assert len(content["bases"]) == len(content["versions"])
-    assert (
-        len(content["runners"])
-        == len(content["names"])
-        == len(content["colors"])
-        == len(content["styles"])
-        == len(content["markers"])
-    )
+def get_longitudinal_plot_config():
+    cfg = mconfig.get_bench_runner_config()
+    if "plot" in cfg:
+        raise ValueError(
+            "It looks like you have an old plot config in bench_runner.toml. "
+            "Please refer to the section 'New Plot Configuration' in docs/CHANGELOG.md "
+            "for detailed migration instructions."
+        )
+
+    plots = cfg.get("longitudinal_plot", {})
+    subplots = plots.get("subplots", [])
+
+    for subplot in subplots:
+        assert "base" in subplot
+        assert "version" in subplot
+        if "flags" not in subplot:
+            subplot["flags"] = []
+
+    return plots
+
+
+@functools.cache
+def get_flag_effect_plot_config():
+    cfg = mconfig.get_bench_runner_config()
+
+    plots = cfg.get("flag_effect_plot", {})
+    subplots = plots.get("subplots", [])
+
+    for subplot in subplots:
+        assert "name" in subplot
+        assert "head_flags" in subplot
+        subplot["head_flags"] = sorted(set(subplot["head_flags"]))
```

Comment:
FWIW, I intend to use this with head_flags set to the empty list, just so I can use the runner_map to compare different compilers on the same host. I don't mind requiring head_flags though, just want to make sure you realise it's useful when head_flags can be empty :)